### PR TITLE
feat(share): #MEX-61 implements sharing world

### PR DESCRIPTION
**Pull request by sjollois**
## Describe your changes
Share the world to manage (open/close world) or to admin (all functions).
## Checklist tests
click share button. add different rights to different people. check if the rights work well
## Issue ticket number and link
[https://jira.support-ent.fr/browse/MEX-61](url)
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by qperie — APPROVED** [review]

File: src/main/java/fr/openent/minetest/Minetest.java [hunk 1]
```diff
@@ -16,17 +21,35 @@
 public class Minetest extends BaseServer {
 	public static final String WORLD_COLLECTION = "world";
 
+	public static final String CONTRIB_RESOURCE_RIGHT = "minetest.contrib";
```
Comment (JinDouang) [inline src/main/java/fr/openent/minetest/Minetest.java]: Est-ce qu'on pourrait créer une classe spécifique pour stocker les magic string qui concerne les droits ?

Genre une classe "Rights" dans le package constant (enum serait mieux mais c pas supporté pour secureAction)

File: src/main/java/fr/openent/minetest/Minetest.java [hunk 2]
```diff
@@ -16,17 +21,35 @@
...
 	@Override
 	public void start() throws Exception {
 		super.start();
 
 		Storage storage = new StorageFactory(vertx, config).getStorage();
 		MinetestConfig minetestConfig = new MinetestConfig(config);
 
+
+		final MongoDbConf conf = MongoDbConf.getInstance();
+		conf.setCollection("world");
```
Comment (JinDouang) [inline src/main/java/fr/openent/minetest/Minetest.java]: magic string

File: src/main/resources/public/sass/global/components/containers/share-world/_share-world.scss [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    button:hover {
+      background-color: $minetest-dark-orange;
+      box-shadow: none;
+      color: $minetest-white;
+      font-size: 14px;
+      margin: 0 2px 10px;
+    }
+
+    button[disabled=disabled] {
+      color: #b1b6bf;
```
Comment (JinDouang) [inline src/main/resources/public/sass/global/components/containers/share-world/_share-world.scss]: variable css pour les couleurs ?

File: src/main/resources/public/sass/global/components/containers/share-world/_share-world.scss [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      background: #8c939e;
```
Comment (JinDouang) [inline src/main/resources/public/sass/global/components/containers/share-world/_share-world.scss]: variable css pour les couleurs ?

**Review by JinDouang — CHANGES_REQUESTED** [review]

**Comment by JinDouang** [comment]
Fais gaffe, tes TUs ne passent plus c'est important de les corriger

File: src/main/resources/public/ts/services/__test__/minetest.service.test.ts [hunk 5]
```diff
@@ -21,6 +21,9 @@ describe('MinetestService', () => {
         const data = {response: true};
 
         const importWorld: IImportWorld = {
+            myRights: undefined,
```
Comment (JinDouang) [inline src/main/resources/public/ts/services/__test__/minetest.service.test.ts]: Je mets ça ici mais attention tes TUs ne passent plus

**Review by JinDouang — DISMISSED** [review]

**Pull request merged**